nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/GridFSBinaryManager.java

```diff
@@ -115,9 +117,18 @@ public void initialize(String blobProviderId, Map<String, String> properties) th
             throw new NuxeoException("Unable to initialize GridFS Binary Manager, properties " + SERVER_PROPERTY
                     + " and " + DBNAME_PROPERTY + " has been removed. Please configure a connection!");
         }
+
+        String namespace = properties.get(NAMESPACE);
         String bucket = properties.get(BUCKET_PROPERTY);
         if (StringUtils.isBlank(bucket)) {
-            bucket = blobProviderId + ".fs";
+            if (StringUtils.isNotBlank(namespace)) {
+                bucket = blobProviderId + "." + namespace.trim();
+            } else {
+                bucket = blobProviderId;
+            }
+            bucket = bucket + ".fs";
+        } else if (StringUtils.isNotBlank(namespace)) {
```

Comment:
I prefer to use `isEmpty` rather than `isBlank` for this sort of property based thing.  Although, I note `isBlank` was already being used.

Reply:
IMHO it's better to test for blank rather than empty because this comes from user input/config and I don't want an XML reformatting like:

    <property name="namespace">
        mynamespace
    </property>

to break things.